vendor/pacman/js/game.js

```diff
@@ -23,52 +23,81 @@ var TIME_FRUITS = 0;
 
 var HELP_DELAY = 1500;
 var HELP_TIMER = -1;
-			
-function blinkHelp() { 
-	if ( $('.help-button').attr("class").indexOf("yo") > -1 ) { 
+
+let gameOver = new Event("gameOver");
+
+window.onload = function() {
+   document.getElementById("panel").addEventListener("gameOver", scorePopUp);
+}
+
+function scorePopUp() {
+	var name = prompt("Please enter your name:", "anonymous");
+	// don't post if person cancels prompt or doesn't enter name value
+	if (name == null || name == "") {
+		SCORE = 0;
+		return;
+	}
+	var score = SCORE;
+	console.log(score);
+	sendScore("/scores/pacman/", {name, score});
+	window.location.href = "/scores/pacman";
```

Comment:
`fetch()` in `sendScore()` is an async function, so since you are immediately changing location after executing the request, in certain cases that request might be stopped prematurely and you could be sent to the high scores page without your score on there. A safer pattern might be to pass a callback into `sendScore()` and use that in `fetch().then()`.
```js
  sendScore("/scores/pacman/", {name, score}, () => {
    window.location.href = "/scores/pacman";
  });
}

function sendScore(url, data, callback) {
  fetch(url, {
    method: 'POST',
    body: JSON.stringify(data),
    headers: {
      'Content-Type': 'application/json',
    },
  }).then(callback);
}
```

Reply:
You can also return the promise from `sendScore`:

```javascript
  sendScore("/scores/pacman/", {name, score}).then(
      () => { window.location.href = "/scores/pacman"; });
}

function sendScore(url, data) {
  return fetch(url, { ...
```

Reply:
Done a218602